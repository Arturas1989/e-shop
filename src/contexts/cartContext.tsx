import { ReactNode, createContext, useContext } from "react";
import { Cart } from "../services/Cart";
import { useCart } from "../hooks";

type CartContextProviderProps = {
  children: ReactNode;
};

type CartContextValue = {
  cart: Cart | null;
  setCart: React.Dispatch<React.SetStateAction<Cart | null>>;
  isLoading: boolean;
} | null;

const CartContext = createContext<CartContextValue>(null);

export const CartContextProvider = ({ children }: CartContextProviderProps) => {
  const {cart, setCart, isLoading} = useCart();
  return (
    <CartContext.Provider value={{cart, setCart, isLoading}}>
      {children}
    </CartContext.Provider>
  )
};

export const useCartContext = () => useContext(CartContext);
