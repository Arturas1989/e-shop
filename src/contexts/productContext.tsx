import { ReactNode, createContext, useContext } from "react";
import { DataInfo, Product } from "../types";
import { useData } from "../hooks";

type ProductContextProviderProps = {
  children: ReactNode;
};

type ProductContextValue = {
  products: Product[] | null;
  setProducts: React.Dispatch<React.SetStateAction<Product[] | null>>;
  isProductsLoading: boolean;
  featuredProducts: Product[] | null;
  setFeatured: React.Dispatch<React.SetStateAction<Product[] | null>>;
  isFeaturedProductsLoading: boolean;
} | null;

const ProductContext = createContext<ProductContextValue>(null);

export const ProductContextProvider = ({ children }: ProductContextProviderProps) => {
  const productsInfo: DataInfo = {
    collectionName: 'products',
    onlyFeatured: false,
  }
  const [products, isProductsLoading, setProducts] = useData<Product>(productsInfo);
  const featuredInfo: DataInfo = {
    collectionName: 'products',
    onlyFeatured: true
  }
  const [featuredProducts, isFeaturedProductsLoading, setFeatured] = useData<Product>(featuredInfo);

  const contextVal = {
    products, isProductsLoading, setProducts, featuredProducts, setFeatured, isFeaturedProductsLoading
  }
  return (
    <ProductContext.Provider value={contextVal}>
      {children}
    </ProductContext.Provider>
  )
};

export const useProductsContext = () => useContext(ProductContext);
