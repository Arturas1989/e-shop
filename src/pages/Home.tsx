export const Home = () => {
  return (
    <main data-testid='home'>Home</main>
  )
};
