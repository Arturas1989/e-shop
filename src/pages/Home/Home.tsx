import { Hero, FeaturedProducts, Testimonials, Faq } from "./index";

export const Home = () => {
  return (
    <main data-testid='home' className="home">
      <Hero />
      <FeaturedProducts />
      <Testimonials />
      <Faq />
    </main>
  )
};
