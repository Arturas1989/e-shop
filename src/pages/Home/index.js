export { Hero } from './components/Hero/Hero';
export { FeaturedProducts } from './components/FeaturedProducts/FeaturedProducts';
export { Testimonials } from './components/Testimonials/Testimonials';