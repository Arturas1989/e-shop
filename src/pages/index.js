export { Home } from './Home/Home';
export { ProductsPage } from './Products/ProductsPage';
export { PageNotFound } from './PageNotFound/PageNotFound';

export { Register } from './Auth/Register/Register';
export { Login } from './Auth/Login/Login';
