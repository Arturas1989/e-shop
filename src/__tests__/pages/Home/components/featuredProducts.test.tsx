import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { FeaturedProducts } from '../../../../pages/Home';
import { ProductContextProvider } from '../../../../contexts/productContext';

describe('Products tests', () => {
  test('FeaturedProducts component renders correctly', () => {
    const tree = renderer
      .create(
        <ProductContextProvider>
          <BrowserRouter>
            <FeaturedProducts />
          </BrowserRouter>
        </ProductContextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});