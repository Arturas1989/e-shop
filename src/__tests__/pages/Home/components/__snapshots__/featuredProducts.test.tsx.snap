// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Products tests FeaturedProducts component renders correctly 1`] = `
<section
  className="feature"
>
  <h2
    className="heading"
  >
    Featured products
  </h2>
  <div
    data-testid="loading"
    style={
      Object {
        "marginTop": "0",
      }
    }
  >
    <img
      alt="spinner"
      src="spinner.gif"
    />
  </div>
</section>
`;
